### DISCUSSION TITLE
per-process in addition to per-app capture.

### DISCUSSION THREAD

**Discussion by bitwraith0** [emanuele-f/PCAPdroid#458] [Ideas]
i tend to work on projects outside of the app_process space. For example, i'm currently integrating a custom adb client into my app. I am in the beginning staged and studying the protocol docs and the adb and adbd code on googlesources. But i also would like to be able to run tests while monitoring traffic between the 2 processes. One of my design goals is to eliminate the need for either to be available on the device.

**Comment by emanuele-f** [comment]
Hello, do you mean TCP/IP traffic happening on localhost? Do you have root? The easiest way should be to run each process with its own unique uid

**Reply by bitwraith0** [reply]
I have root. i'm using pcapdroid to double-check/debug my interaction with adb /adbd .. At this time it's all localhost traffic.

I'd like to be able to select the pid vs. app.
make sense? maybe I can accomplish a different way.

**Reply by emanuele-f** [reply]
I see. Filtering traffic by pid is not trivial, not only in Android but in linux in general

**Reply by bitwraith0** [reply]
yessir. understood.
